### PR TITLE
setup init

### PR DESCRIPTION
**Pull request by PavelCherniavskyi**

### CODE REVIEW

File: tests/test_cases.cpp [hunk 1]
```diff
@@ -0,0 +1,11 @@
+#include <gtest/gtest.h>
+#include <spdlog/spdlog.h>
+
+TEST(SharedPtr, SharedPtrTest)
+{
+  const int a = 5;
+  const int b = 5;
+  EXPECT_EQ(a, b);
+
+  SPDLOG_INFO("Hello from SPDLOG: {}-{}", a, b);
```
Comment (vnikolai) [inline tests/test_cases.cpp]: SPDLOG does not seems necessary

File: tests/gtest_main.cpp [hunk 2]
```diff
@@ -0,0 +1,10 @@
+#include <gtest/gtest.h>
+#include <spdlog/spdlog.h>
+
+int main(int argc, char **argv) 
+{
+  spdlog::set_pattern("[%^-%L-%$][%S.%e][%s:%# %!] %v");
+
+  ::testing::InitGoogleTest(&argc, argv);
+  return RUN_ALL_TESTS();
+}
```
Comment (vnikolai) [inline tests/gtest_main.cpp]: preferred newlines at the eof

File: CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+conan_cmake_autodetect(settings)
+
+conan_cmake_install(PATH_OR_REFERENCE .
+                    BUILD missing
+                    REMOTE conancenter
+                    SETTINGS ${settings})
+
+find_package(GTest)
+find_package(spdlog)
```
Comment (vnikolai) [inline CMakeLists.txt]: spdlog does not  seem necessary

File: main.cpp [hunk 4]
```diff
@@ -0,0 +1,14 @@
+#include <spdlog/spdlog.h>
```
Comment (vnikolai) [inline main.cpp]: why is this file needed?

File: src/shared_ptr.hpp [hunk 5]
```diff
@@ -0,0 +1,12 @@
+#ifndef SHARED_PTR_HPP
```
Comment (vnikolai) [inline src/shared_ptr.hpp]: please use #pragma once, this is  supported by all compilers

File: CMakeLists.txt [hunk 6]
```diff
@@ -0,0 +1,44 @@
+cmake_minimum_required(VERSION 3.15)
+
+project(shared_ptr)
+
+list(APPEND CMAKE_MODULE_PATH ${CMAKE_BINARY_DIR})
+list(APPEND CMAKE_PREFIX_PATH ${CMAKE_BINARY_DIR})
+
+add_definitions("-std=c++17")
+
+if(NOT EXISTS "${CMAKE_BINARY_DIR}/conan.cmake")
```
Comment (vnikolai) [inline CMakeLists.txt:10]: is conan necessary?

File: .clang-format [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+ContinuationIndentWidth: 2
+Cpp11BracedListStyle: false
+DerivePointerAlignment: false
+DisableFormat: false
+ExperimentalAutoDetectBinPacking: true
+FixNamespaceComments: true
+IndentCaseLabels: false
+IndentWidth: 2
+IndentWrappedFunctionNames: true
+JavaScriptQuotes: Leave
```
Comment (vnikolai) [inline .clang-format]: Java and Objective C references might be redundant here

File: main.cpp (hunk 4 above)
Comment (PavelCherniavskyi) [reply main.cpp]: For spd_log. I will remove that.

File: src/shared_ptr.hpp (hunk 5 above)
Comment (PavelCherniavskyi) [reply src/shared_ptr.hpp]: What about that?
https://ru.stackoverflow.com/questions/630573/ifndef-define-vs-pragma-once

File: CMakeLists.txt (hunk 6 above)
Comment (PavelCherniavskyi) [reply CMakeLists.txt:10]: Of course not. But it is handy way to build project and link library out of the box without explicit installation third parties.

File: src/shared_ptr.hpp (hunk 5 above)
Comment (vnikolai) [reply src/shared_ptr.hpp]: That article has no valid points in it

File: CMakeLists.txt (hunk 6 above)
Comment (vnikolai) [reply CMakeLists.txt:10]: Separate installation is known practice, generally used inside linux ecosystem and currently starts to be supported on microsoft as well. Please feel free to keep your GTest delivery mechanism if it is comfortable for you

File: main.cpp [hunk 8]
```diff
@@ -0,0 +1,9 @@
+#include "src/shared_ptr.hpp"
```
Comment (vnikolai) [inline main.cpp]: still not sure if this executable is needed at all

File: tests/gtest_main.cpp [hunk 9]
```diff
@@ -0,0 +1,7 @@
+#include <gtest/gtest.h>
```
Comment (vnikolai) [inline tests/gtest_main.cpp]: is this file necessary?

File: CMakeLists.txt [hunk 10]
```diff
@@ -0,0 +1,37 @@
...
+conan_cmake_install(PATH_OR_REFERENCE .
+                    BUILD missing
+                    REMOTE conancenter
+                    SETTINGS ${settings})
+
+find_package(GTest)
+
+add_subdirectory(tests)
+
+add_executable(${PROJECT_NAME} main.cpp)
```
Comment (vnikolai) [inline CMakeLists.txt]: still not sure if this executable is needed at all

**Review by vnikolai — APPROVED** [review]
